source/administrator-guide/managing-users-and-groups/authentication-mode.rst

```diff
@@ -510,6 +511,291 @@ You can configure your environment by updating the previous file or by defining
     cas.login.url=${cas.baseURL}/login
     cas.logout.url=${cas.baseURL}/logout?url=${geonetwork.https.url}/
 
+
+.. _authentication-openid:
+
+Configuring OAUTH2 OpenID Connect
+---------------------------------
+
+`OAUTH2 OpenID Connect <https://openid.net/connect/>`_ is an authentication and authorization system based on OAUTH2.  Geonetwork's OpenID Connect plugin has been tested with `Keycloak <https://keycloak.org>`_ and `Azure AD <https://azure.microsoft.com/en-ca/services/active-directory/>`_, but should work with any provider.
+
+Basic Setup Steps;
+
+#. Configure your IDP Server (i.e. Keycloak or Azure AD)
+
+    #. Ensure that the ID Token provides role/group information
+    #. Authorize your Geonetwork URLs for redirects (i.e. `http://localhost:8080/geonetwork/login/oauth2/code/geonetwork-oicd`) 
+    #. Record the Client ID 
+    #. Record the Client Secret
+    #. Get the Server's JSON metadata document
+
+#. Configure Geonetwork via environment variables
+    
+        #. `GEONETWORK_SECURITY_TYPE=openidconnect`
+        #. `OPENIDCONNECT_CLIENTSECRET=...`   (from your IDP server)
+        #. `OPENIDCONNECT_CLIENTID=...`  (from your IDP server)
+        #. `OPENIDCONNECT_SERVERMETADATA_JSON_TEXT='...'` (the text of your Server's JSON metadata document)
+        #. `OPENIDCONNECT_IDTOKENROLELOCATION=...` (location of the user's roles in the ID Token)
+
+
+
+Geonetwork's Open ID Connect plugin has a lot of configuration options - please see the `config-security-openidconnect.xml` and `config-security-openidconnect-overrides.properties` files.
+
+
+Environment Variable and Meaning
+================================
+
+**GEONETWORK_SECURITY_TYPE**
+
+    Should be `openidconnect`.
+
+**OPENIDCONNECT_CLIENTID**
+
+    The name of the client/application you configured on your OpenID server.    
+
+**OPENIDCONNECT_CLIENTSECRET**
+
+    The `client secret` you configured on your OpenID server.  
+
+**OPENIDCONNECT_SERVERMETADATA_JSON_TEXT**
+
+    Should be the text of your OpenID server's metadata configuration (JSON).
+
+**OPENIDCONNECT_SERVERMETADATA_FNAME**
+
+    Instead of putting the OpenID server's metadata configuration as text in a variable (`OPENIDCONNECT_SERVERMETADATA_JSON_TEXT`), you can put the JSON contents in a file and reference it with this variable (ie. `/WEB-INF/config-security/openid-configuration.json`)
```

Comment:
Better to use to format /WEB-INF/config-security/openid-configuration.json `` instead of `

Reply:
changed